[StaleBot] Add first version of stale.yml config.

sunshinejr opened the pull request:
Rationale: #1071. 

~~(I'm gonna comment in the PR my opinions, bear with me for a sec :D)~~
**Edit:** Finished my feedback, I'd love to hear your opinion on this!

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+# Configuration for probot-stale - https://github.com/probot/stale
+
+# Number of days of inactivity before an Issue or Pull Request becomes stale
+daysUntilStale: 60
```
In my opinion this is the most reasonable - one month is too short, 3 months kinda too long?

sunshinejr replied:
I would agree if we had that much Issues, but we don't, really. On the first page we have an issue from December last year, so in my opinion 10/30 will be too short.

AndrewSB replied:
That's true, but if I think of issues that are opened now, (to me) it feels like an issue without a response for a month feels stale

I would love to hear what other people think - @BasThomas especially (since you handle closing so many of our current stale issues 😋)

Totally fine with 60 otherwise, Im just trying to make sure our bot is useful

sunshinejr replied:
So, how are we feeling about this one? @AndrewSB, @pedrovereza, @BasThomas

BasThomas replied:
Let's just go with 45 in that case and adjust if needed?

pedrovereza replied:
I'm ok going with what we have now. We probably won't get to the optimal number before we see the bot running and judge how useful it's being :wink:

AndrewSB replied:
I'm down to go with 45 and then adjust in the future

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+daysUntilStale: 60
+
+# Number of days of inactivity before a stale Issue or Pull Request is closed
+daysUntilClose: 14
```
On the other hand, if we introduce this bot now, we might want to give more time for users to respond to issues. Thus, making me change the default of 7 days to 14 for days until close. I think we may adjust the values as we go, but at the beginning it might be better to start off high.

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+
+# Issues or Pull Requests with these labels will never be considered stale
+exemptLabels:
+  - "enhancement"
```
Wasn't sure if this should be the only one white-listed labels, thoughts?

pedrovereza replied:
We may want to include `chore` as well 🤔

AndrewSB replied:
as well as `confirmed bug` those often take a while to be resolved

sunshinejr replied:
I agree! Both `chore` & `confirmed bug` are a great additions.

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+  - "enhancement"
+
+# Label to use when marking as stale
+staleLabel: "awaiting input"
```
And also wasn't sure with naming in this one, took the default here.

AndrewSB replied:
LGTM

BasThomas replied:
What about just "stale"? I can't really think of anything better, although it seems like something's just on the tip of my tongue. "awaiting input" seems... off.

BasThomas replied:
"no recent activity" maybe?

sunshinejr replied:
I really like `no recent activity`. Gonna change to that one!

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+staleLabel: "awaiting input"
+
+# Comment to post when marking as stale. Set to `false` to disable
+markComment: >
```
This is also default comment, someone with better wording skills might want to take a stab at this one :D

BasThomas replied:
"This issue has been marked as stale because it has not had any recent activity. If no further activity occurs, it will be closed. Please feel free to open a new issue / Please feel free to reopen this issue if needed."

At least the "automatically" seems redundant. Also, for issues, "Thank you for you contributions" seems strange.

BasThomas replied:
It would be nice to thank them for taking the time to file an issue, though.

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+  for your contributions.
+
+# Comment to post when removing the stale label. Set to `false` to disable
+unmarkComment: false
```
I think it's meaningless to have a comment that says "label was removed", since it will be seen after any comment appears in the issue. This is just my opinion, though.

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+# Comment to post when closing a stale Issue or Pull Request. Set to `false` to disable
+closeComment: >
+  This issue will be auto-closed because there hasn't been any activity for a few months.
+  Feel free to [open a new one](https://github.com/Moya/Moya/issues/new) if you still experience this problem :+1:
```
Here, I was contemplating if suggestion to create a new issue will be better than reopening current one. I might like latter, but I'd love to hear your opinions.

pedrovereza replied:
Would it better the specify the exact number of days we waited for a response?

AndrewSB replied:
I think it would be, and also add a sentence saying something along the lines of "please open a new issue to further discuss the issue, if desired"

BasThomas replied:
I'm not sure on reopening vs opening a new one. On wording: "This issue has been auto-closed [..]" I think would better reflect the action.

pedrovereza replied:
I'm ok with reopening, creating a new one would probably be just duplication.

AndrewSB replied:
hmm, we wouldn't be able to lock the issue then.
Should we keep locking as a manual thing, not something we do to stale issues?

I'm just afraid of someone responding to a closed issue later, and one of us having to pay attention to that and re-opening the issue.
I feel as if that goes against what we're trying to automate away with this bot

sunshinejr replied:
+1 to creating a new one just because of locking, so let's stay with what we've got ^^

sunshinejr commented on .github/stale.yml:
```diff
@@ -0,0 +1,31 @@
+  Feel free to [open a new one](https://github.com/Moya/Moya/issues/new) if you still experience this problem :+1:
+
+# Limit to only `issues`
+only: issues
```
Also, wasn't sure here, but I've figured, that we don't have a problem with PRs right now, but more with Issues.

BasThomas commented:
Maybe not for now, but I was thinking: can we lock issues with x comments after being closed?

sunshinejr commented:
Looking at the [lib file](https://github.com/probot/stale/blob/master/lib/stale.js), I think there is no `lock()` method and any other that are needed for this to happen. Although I love the idea and we can try to add an Issue or PR if someone is more skilled in .js than me 😅

AndrewSB commented:
@BasThomas read my mind. Locking would be awesome, let's create that issue 😄

BasThomas commented:
GitHub apparently doesn't support locking (yet)... https://github.com/probot/stale/issues/16#issuecomment-291862121

BasThomas commented:
... but there seems to be a plugin that *does* work? I can't really find out how this would work: https://github.com/lee-dohm/lock.

BasThomas commented:
Thanks for all this work though, @sunshinejr! This seems like a great addition. 🎉

sunshinejr commented:
I've updated the PR:
- Added `confirmed bug` & `chore` in exempt labels.
- Renamed stale label to `no recent acitivity`
- Changed the comments for marking & closing.

Let me know what do you guys think :)

sunshinejr commented:
Should be good to go 👍
